pkg/engine/runtime/terraform/util.go

```diff
@@ -0,0 +1,163 @@
+package terraform
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/hashicorp/hc-install/product"
+	"github.com/hashicorp/hc-install/releases"
+	apiv1 "kusionstack.io/kusion/pkg/apis/core/v1"
+	"kusionstack.io/kusion/pkg/util/kfile"
+	"kusionstack.io/kusion/pkg/util/pretty"
+)
+
+var (
+	tfInstallSubDir  = "terraform"
+	tfInstallTimeout = 3 * time.Minute
+)
+
+type InstallOption struct {
+	MutePrompt bool
+	Intent     *apiv1.Intent
+}
+
+// check and install the terraform executable binary if it has not been downloaded.
+func (io *InstallOption) CheckAndInstall() error {
+	if len(io.Intent.Resources) < 1 {
+		return nil
+	}
+
+	for i, res := range io.Intent.Resources {
+		if res.Type == apiv1.Terraform {
+			break
+		}
+
+		if i == io.Intent.Resources.Len() {
+			return nil
+		}
+	}
+
+	sp := &pretty.SpinnerT
+	if err := io.checkTerraformExecutable(); err != nil {
+		if !io.MutePrompt {
+			sp.WarningPrinter.Printfln("Terraform executable binary is not found")
+		}
+
+		if err := io.installTerraform(); err != nil {
+			return err
+		}
+
+		if !io.MutePrompt {
+			sp.SuccessPrinter.Printfln("Successfully installed terraform and set the executable path")
```

Comment:
I am not sure whether it's a good chose to print terrform related words in the console, how does radius work?

Reply:
Radius prints relevant information in the log, and I have updated the implementation here.